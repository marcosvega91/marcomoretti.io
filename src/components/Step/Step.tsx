import React, { ReactText } from 'react';
import styled, { css } from 'styled-components';

export interface StepProps {
  index?: number;
  active?: boolean;
  children?: ReactText;
}

const StepContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const stepActive = css`
  background-color: black;
  color: white;
`;

const stepInactive = css`
  background-color: white;
  border: 1px solid black;
  color: black;
`;

const StepIndexContainer = styled.div<{ active: boolean }>`
  width: 24px;
  height: 24px;
  border-radius: 12px;
  ${({ active }) => (active ? stepActive : stepInactive)});
  line-height: 24px;
  text-align: center;
  color: white;
`;

const StepLabel = styled.span`
  color: black;
`;

const Step = ({ index, children, active = false }: StepProps) => (
  <StepContainer>
    <StepIndexContainer active={active}>{index}</StepIndexContainer>
    <StepLabel>{children}</StepLabel>
  </StepContainer>
);

export default Step;
