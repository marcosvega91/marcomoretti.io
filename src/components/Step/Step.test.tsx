import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Step from './index';

describe('<Step />', () => {
  it('should show index 1 if no index is passed', () => {
    const { container } = render(<Step />);
    const step = container.firstChild;
    const indexContainer = step?.firstChild;
    expect(indexContainer).toHaveTextContent('1');
  });

  it('should show have different backgroundColor depending on active prop', () => {
    const { container, rerender } = render(<Step />);
    const step = container.firstChild;
    const indexContainer = step?.firstChild;
    expect(indexContainer).toHaveStyle(`
      backgroundColor: 'white',
    `);
    rerender(<Step active />);
    expect(indexContainer).toHaveStyle(`
      backgroundColor: 'black',
    `);
  });

  it('should call onStepSelect clicking on the step passing index as param', () => {
    const onStepSelect = jest.fn();
    const { container } = render(<Step onStepSelect={onStepSelect} />);
    const step = container.firstChild;
    expect(step).not.toBeNull();
    if (step) {
      fireEvent.click(step);
    }
    expect(onStepSelect.mock.calls.length).toEqual(1);
    expect(onStepSelect.mock.calls[0][0]).toEqual(0);
  });

  it('should render children in a span', () => {
    const { container } = render(<Step>Test</Step>);
    const label = container.querySelector('span');
    expect(label).toHaveTextContent('Test');
  });

  it('should not render span if not children is passed', () => {
    const { container } = render(<Step />);
    const label = container.querySelectorAll('span');
    expect(label.length).toEqual(0);
  });
});
