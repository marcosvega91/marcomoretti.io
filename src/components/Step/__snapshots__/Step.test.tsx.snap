// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Step /> should match snapshot when active = false 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1 {
  width: 24px;
  height: 24px;
  border-radius: 12px;
  background-color: white;
  border: 1px solid black;
  color: black;
  line-height: 24px;
  text-align: center;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    1
  </div>
</div>
`;

exports[`<Step /> should match snapshot when active = true 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1 {
  width: 24px;
  height: 24px;
  border-radius: 12px;
  background-color: black;
  color: white;
  line-height: 24px;
  text-align: center;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    1
  </div>
</div>
`;
